cli/version/version_match.go

```diff
@@ -0,0 +1,174 @@
+package version
+
+import (
+	"fmt"
+
+	"github.com/tarantool/tt/cli/util"
+)
+
+type requiredField uint8
+type requiredFields []requiredField
+
+const (
+	// Values according order of Fields in regex from `matchVersionParts()`.
+	requiredFieldBuildName requiredField = iota
+	requiredFieldMajor
+	requiredFieldMinor
+	requiredFieldPatch
+	requiredFieldReleaseType
+	requiredFieldReleaseNum
+	requiredFieldAdditional
+	requiredFieldHash
+	requiredFieldRevision
+
+	countOfRequiredFields
+)
+
+// NotFoundError is sentinel error for case when required version not found.
+type NotFoundError struct {
+	expected string
+}
+
+func (e NotFoundError) Error() string {
+	return fmt.Sprintf("not matched with: %s", e.expected)
+}
+
+// exploreMatchVersion examines the given string and retrieves the version components,
+// that define the rules for checking for version consistency.
+func exploreMatchVersion(verStr string) (Version, requiredFields, error) {
+	var version Version
+	fields := make(requiredFields, 0, countOfRequiredFields)
+
+	matches, err := matchVersionParts(verStr, false)
+	if err != nil {
+		return version, fields, err
+	}
+
+	version.BuildName = matches["buildname"]
+	if version.BuildName != "" {
+		fields = append(fields, requiredFieldBuildName)
+	}
+
+	if matches["major"] != "" {
+		if version.Major, err = util.AtoiUint64(matches["major"]); err != nil {
+			return version, fields, fmt.Errorf("can't parse Major: %w", err)
+		}
+		fields = append(fields, requiredFieldMajor)
+	}
+
+	if matches["minor"] != "" {
+		if matches["major"] == "" {
+			return version, fields, fmt.Errorf("parse Minor requires Major: %w", err)
```

Comment:
Something wrong with the output, the error is nil:

```bash
$ ./tt install tt .2.1 
   • Searching in versions... 
   ⨯ parse Minor requires Major: %!!(MISSING)w(<nil>)
```

Reply:
Yeah, here there is no `err` value